pkg/scheduler/core/scheduling_queue.go

```diff
@@ -406,6 +422,71 @@ func (p *PriorityQueue) WaitingPodsForNode(nodeName string) []*v1.Pod {
 	return pods
 }
 
+// Run starts the goroutine to pump from podBackoff to activeQ
+func (p *PriorityQueue) Run() {
+	go wait.Until(p.moveNextBackoffToActive, 0, p.stop)
```

Comment:
Why is this method better than checking backoff time when a pod needs to be moved from unschedulable queue to active queue and if backoff time is larger than zero, we start a go routine that sleeps (by calling `backoff_util.TryWait`) and then moves the pod to the active queue. Your logic looks more complicated to me.

Reply:
My thinking was that spawning a goroutine potentially per re-scheduled pod could be a scaling issue. I agree that it would be much simpler to do this though, so I'm happy to be told this isn't an issue.